exercise_perseus_render/assets/src/vue/perseus.vue

```diff
@@ -0,0 +1,235 @@
+
+      // Add in a couple of addons that Perseus needs.
+      global.React.addons = global.React.__internalAddons = {
+        CSSTransitionGroup: require('react-addons-css-transition-group'),
+        PureRenderMixin: require('react-addons-pure-render-mixin'),
+      };
+
+      // Perseus also expects katex to be globally imported.
+      global.katex = require('perseus/lib/katex/katex');
+
+      // Underscore as well! We use their bundled version for compatibility reasons.
+      global._ = require('perseus/lib/underscore');
+
+      // Take note of the global '$' variable so we can replace it after we remove jQuery
+      this.backup$ = global.$;
+
+      // Load in jQuery, because apparently we still need that for a React app.
+      require('perseus/lib/jquery');
+
+      // Perseus expects this i18n object, but hopefully we won't have to touch it
+      // We should try to only use our interface text, so as to avoid interacting with this.
+      require('perseus/lib/i18n');
+
+      // For reasons quite beyond my ken, some configuration is still delegated to this
+      // global Exercises object.
+      this.Exercises = global.Exercises = {
+        useKatex: true,
+      };
+
+      this.perseus = require('perseus/build/perseus');
+
+      // A handy convenience mapping to what is essentially a constructor for Item Renderer
+      // components.
+      this.itemRendererFactory = this.react.createFactory(this.perseus.ItemRenderer);
+    },
+
+    destroyed() {
+      // Clean up the global namespace pollution that Perseus necessitates.
+      delete global.React;
+      delete global.ReactDOM;
+      global.$ = this.backup$;
+      delete global.i18n;
+      delete global.Exercises;
+    },
+
+    $trNameSpace: 'perseus',
+    $trs: {
+      showScratch: 'Show scratchpad',
+      notAvailable: 'Scratchpad not available',
+      loading: 'Loading',
+      check: 'Check Answer',
+      correct: 'Correct! Next Question.',
+      incorrect: 'Incorrect, try again.',
+      hint: 'Hint',
+    },
+    props: {
+      scratchpad: {
+        type: Boolean,
+        default: false,
+      },
+      initialHintsVisible: {
+        type: Number,
+        default: 0,
+      },
+      item: {
+        type: Object,
+        validator: (obj) => [
+          // A somewhat protracted validator to ensure that our item data conforms
+          // to that expected by the Perseus ItemRenderer,
+          // c.f. https://github.com/Khan/perseus/blob/master/src/item-renderer.jsx#L35
+          'calculator',
+          'chi2Table',
+          'periodicTable',
+          'tTable',
+          'zTable',
+        ].reduce(
+            // Loop through all of the above properties and ensure that if the 'answerArea'
+            // property of the item has them, then their values are set to Booleans.
+            (prev, key) => !(!prev ||
+              obj.answerArea.hasOwnProperty(key) &&
+              typeof obj.answerArea[key] !== 'boolean'), true) &&
```

Comment:
is this reduce function try to check that obj.answerArea does not contain anything from the list `['calculator', 'chi2Table', ...]` ?